pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/stream/ReadableStreamingData.java

```diff
@@ -151,14 +151,16 @@ public byte readByte() {
     /** {@inheritDoc} */
     @Override
     public long skip(final long n) {
-        final long clamped = Math.min(n, limit);
+        if (n > limit) {
+            throw new BufferUnderflowException();
+        }
 
-        if (clamped <= 0) {
+        if (n <= 0) {
             return 0;
         }
 
         try {
-            long numSkipped = in.skip(clamped);
+            long numSkipped = in.skip(n);
```

Comment:
Does it make sense to check that `numSkipped` is equal to `n`?

Reply:
Interesting... The method is allowed to skip less bytes than requested and return the number of actually skipped bytes. However, it might be only because the method never tried to skip past input stream limit. Now method contract is going to be changed (I assume), so if n is greater than the limit, then an exception is thrown. In this case, the method will be guaranteed to always skip the requested number of bytes or throw an exception, right? In this case, if `in.skip(n)` skipped less than `n`, it must result in an exception, too. What do you think?

Reply:
This seems to make sense in theory. However, the InputStream.skip() doc says:

```
The skip method may, for a variety of reasons, end up skipping over some smaller number of bytes, possibly 0. This may result from any of a number of conditions; reaching end of file before n bytes have been skipped is only one possibility.
```

So it may skip less than `n` bytes for a random, not necessarily bad reason.

So I'm unsure if we should error out on this, given that we honestly return the number of bytes actually skipped, allowing the caller to make its own decision.

Please let me know what you think.